fill/fill_station.cc

```diff
@@ -153,9 +173,22 @@ class RocketTelemeterServiceImpl final : public RocketTelemeter::Service
                     // Broken stream
                     return Status::CANCELLED; 
                 }
+
+                errorTimerStarted = false;
             } else {
                 spdlog::error("Error reading the full packet with message:\n");      
-                std::cout << t.status() << std::endl;           
+                std::cout << t.status() << std::endl;     
+
+                // starts timer 
+                if (!errorTimerStarted) { 
+                    errorStartTime = std::chrono::steady_clock::now();
+                    errorTimerStarted = true;
+                } else {
+                    if (std::chrono::steady_clock::now() - errorStartTime >= errorThreshold) {
+                        spdlog::warn("Could not reconnect to rocket. Closing the ball valve.\n");
```

Comment:
Why can't we just close the BV manually in this case?

We'd still have control over the fill station and knowledge that the rocket is disconnected (or at least we should)

Reply:
I think it would be cleaner to have a second watchdog here, but if this works, feel free to keep it

Reply:
Yeah closing BV manually would prob be safer tbh, good point